reference/build_helpers/cmake.rst

```diff
@@ -292,3 +292,38 @@ Environment variables
 
 There are some environment variables that will also affect the ``CMake()`` helper class. Check them in the
 :ref:`CMAKE RELATED VARIABLES<cmake_related_variables>` section.
+
+Example
+-------
+The following example of ``conanfile.py`` shows you how to manage a project with conan and CMake.
+
+.. code-block:: python
+
+    from conans import ConanFile, CMake
+
+    class SomePackage(ConanFile):
+        name = "SomePackage"
+        version = "1.0.0"
+        settings = "os", "compiler", "build_type", "arch"
+        generators = "cmake"
+
+    def configure_cmake(self):
+        cmake = CMake(self)
+
+        # put definitions here so that they are re-used in cmake between
+        # build(), test() and package()
+        cmake.definitions["SOME_DEFINITION_NAME"] = "On"
+
+        return cmake
+
+    def build(self):
+        cmake = self.configure_cmake()
+        cmake.build()
+
+    def test(self):
```

Comment:
This is good for ``test_package`` functionality, but this code is not run for normal packaging.
Maybe many users want to make sure to run the tests in the ``build()`` method, so they are run for every build of the package, and abort the packaging if they fail. In that case, the ``test()`` method will be removed, and the build would be:

```python
   def build(self):
        cmake = self.configure_cmake()
        cmake.build()
        cmake.test()
```

Reply:
I added the example commands for two scenarios: one that you suggest (run unit tests after the build), and another command to run a custom make command.